File: rb/features.rb
# HanoSnapchatLens SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module HanoSnapchatLensFeatures
  def self.make_feature(name)
    case name
    when "base"
      HanoSnapchatLensBaseFeature.new
    when "ratelimit"
      HanoSnapchatLensRatelimitFeature.new
    when "retry"
      HanoSnapchatLensRetryFeature.new
    when "test"
      HanoSnapchatLensTestFeature.new
    when "timeout"
      HanoSnapchatLensTimeoutFeature.new
    else
      HanoSnapchatLensBaseFeature.new
    end
  end
end

File: py/features.py
# HanoSnapchatLens SDK feature factory

from feature.base_feature import HanoSnapchatLensBaseFeature
from feature.test_feature import HanoSnapchatLensTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HanoSnapchatLensBaseFeature(),
        "test": lambda: HanoSnapchatLensTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
